sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/InMemoryPartitionManager.java

```diff
@@ -20,30 +20,52 @@
     private final Map<String, PartitionOwnership> partitionOwnershipMap = new ConcurrentHashMap<>();
     private final ClientLogger logger = new ClientLogger(InMemoryPartitionManager.class);
 
+    /**
+     * {@inheritDoc}
+     *
+     * @param eventHubName The Event Hub name to get ownership information.
+     * @param consumerGroupName The consumer group name.
+     * @return A {@link Flux} of partition ownership information.
+     */
     @Override
     public Flux<PartitionOwnership> listOwnership(String eventHubName, String consumerGroupName) {
         logger.info("Listing partition ownership");
         return Flux.fromIterable(partitionOwnershipMap.values());
     }
 
+    /**
+     * Returns a {@link Flux} of partition ownership details for successfully claimed partitions. If a partition is
+     * already claimed by an instance or if the eTag in the request doesn't match the previously stored eTag, then
+     * ownership claim is denied.
+     *
+     * @param requestedPartitionOwnerships Array of partition ownerships this instance is requesting to own.
+     * @return Successfully claimed partition ownerships.
+     */
     @Override
     public Flux<PartitionOwnership> claimOwnership(PartitionOwnership... requestedPartitionOwnerships) {
         return Flux.fromArray(requestedPartitionOwnerships)
             .filter(partitionOwnership -> {
                 return !partitionOwnershipMap.containsKey(partitionOwnership.partitionId())
-                    || partitionOwnershipMap.get(partitionOwnership.partitionId()).eTag().equals(partitionOwnership.eTag());
+                    || partitionOwnershipMap.get(partitionOwnership.partitionId()).eTag()
+                    .equals(partitionOwnership.eTag());
             })
             .doOnNext(partitionOwnership -> logger
                 .info("Ownership of partition {} claimed by {}", partitionOwnership.partitionId(),
                     partitionOwnership.ownerId()))
             .map(partitionOwnership -> {
-                partitionOwnership.eTag(UUID.randomUUID().toString()); // set new etag
-                partitionOwnership.lastModifiedTime(System.currentTimeMillis());
+                partitionOwnership.eTag(UUID.randomUUID().toString())
+                    .lastModifiedTime(System.currentTimeMillis());
                 partitionOwnershipMap.put(partitionOwnership.partitionId(), partitionOwnership);
                 return partitionOwnership;
             });
     }
 
+    /**
+     * Updates the checkpoint with new sequence number and offset. A new eTag is generated when the update is successful.
+     *
+     * @param checkpoint Checkpoint information containing sequence number and offset to be stored for this partition.
+     * @return A new eTag associated with the updated checkpoint.
+     */
     @Override
     public Mono<String> updateCheckpoint(Checkpoint checkpoint) {
```

Comment:
You'll have to evaluate this through the lens of Java idioms, but this feels like a poor abstraction to me.  I would suggest taking only the fields that you care about rather than taking the full object and meticulously documenting what you read from it.  Is there value in having a checkpoint instance over the sequence number and offset?

Reply:
This is the Partition Manager that is responsible for storing and updating checkpoint in the backing store (in this case - in-memory) which includes a sequence number, offset, the event processor id, partition id, Event Hub name and consumer group name. Just the sequence number and offset will not be sufficient for a partition manager. If you are, however, referring to CheckpointManager, then yes, it is sufficient to have just sequence number and offset and that's one of the method overloads we have in CheckpointManager. 

This is similar to JS and Python:
https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/src/inMemoryPartitionManager.ts#L53
https://github.com/Azure/azure-sdk-for-python/blob/eventhubs_preview2/sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/partition_manager.py#L39

Is .NET using a different approach?

Reply:
I can't speak to what the .NET client is doing off the top of my head.   I've no issue with accepting the checkpoint if that's what is needed so long as we're not directly modifying the members of the parameter to return results.   _(which I really can't tell if we're doing here or not.)_     In this case, I'd say that the point of contention is likely the comment as it specifically calls out two attributes that are updated and implies that "the checkpoint" is being updated.    When you look at that in combination with the fact that the method accepts a parameter called `checkpoint`, I think that it is misleading and doesn't offer a clear contract for callers.   If we are directly updating the checkpoint, then I would prefer to see us return a new instance rather than modify the calling instance.   If we're set of in-place updates, then I believe the comment needs to make very, very clear to callers that we're going to mutate the parameter.

Reply:
Makes sense. The issue is with the javadoc not being clear. I have updated the javadoc to make it clear that the update will happen in the storage (in-memory) by using the `checkpoint` information provided to this method.